### PR TITLE
Add support for client and self-signed TLS certificates

### LINKED ISSUE #32: Support client connections to TLS servers
The Consul HTTP API might be secured using TLS, the consul_exporter should support that in order to query for data.

### PR DESCRIPTION
**Pull request by grobie**
It's now possible to specify a custom CA certificate to validate the
authenticity of a self-signed server certificate and also to configure a
client certificate with the corresponding key file so that the server
can verify the authenticity of the exporter itself.

The new flags are called:

    # CA certificate which signed the server certificate.
    -consul.ca-file=/path/to/ca.cert
    # Files of the client/exporter certificate.
    -consul.cert-file=/path/to/client.cert
    -consul.key-file=/path/to/client.key

Fixes #32.

@brian-brazil @soumyadipDe @deepthawtz

### CODE REVIEW

File: consul_exporter.go [hunk 1]
```diff
@@ -96,10 +104,22 @@ func NewExporter(uri, kvPrefix, kvFilter string, healthSummary bool, consulTimeo
...
 
+	tlsConfig, err := consul_api.SetupTLSConfig(&consul_api.TLSConfig{
+		CAFile:   opts.caFile,
+		CertFile: opts.certFile,
+		KeyFile:  opts.keyFile,
+	})
+	if err != nil {
+		return nil, err
+	}
+	transport := cleanhttp.DefaultPooledTransport()
```
Comment (brian-brazil) [inline consul_exporter.go]: This disables keepalives, given we've had reports of consul running out of FDs that sounds like a bad idea.

Comment (brian-brazil) [inline consul_exporter.go]: Ignore me, was looking at the wrong function.

**Comment by grobie** [comment]
I was looking at how other Prometheus components handle TLS connections and could only find the blackbox_exporter and prometheus server, are there others? As the consul_exporter can leverage the consul API package to do all the actual TLS work, it didn't seem to help too much using the config package in common. Consul seems to use a well defined set of TLS options and I don't expect there will be a need to add more parameters (except maybe setting a custom server name) https://www.consul.io/docs/agent/encryption.html. I had the feeling splitting out three config options into a file while having the rest as flags would make the configuration more complicated. Using flags seems to be the more common option to configure exporters after all.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**